engine/kafka/src/main/scala/pl/touk/nussknacker/engine/kafka/validator/TopicsExistenceValidator.scala

```diff
@@ -0,0 +1,107 @@
+package pl.touk.nussknacker.engine.kafka.validator
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.scalalogging.LazyLogging
+import pl.touk.nussknacker.engine.kafka.KafkaConfig
+import pl.touk.nussknacker.engine.util.cache.SingleValueCache
+
+import java.util.Properties
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import org.apache.kafka.clients.admin.{Admin, AdminClient, DescribeClusterOptions, DescribeConfigsOptions, ListTopicsOptions}
+import org.apache.kafka.common.config.ConfigResource
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.CustomNodeError
+import pl.touk.nussknacker.engine.kafka.validator.CachedTopicsExistenceValidatorConfig._
+
+import scala.collection.JavaConverters._
+
+trait TopicsExistenceValidator {
+
+  final def validateTopic(topic: String): Validated[TopicExistenceValidationException, String] = validateTopics(List(topic)).map(_.head)
+
+  def validateTopics(topics: List[String]): Validated[TopicExistenceValidationException, List[String]]
+}
+
+trait WithCachedTopicsExistenceValidator extends TopicsExistenceValidator {
+  protected val kafkaConfig: KafkaConfig
+  protected val validatorConfig: CachedTopicsExistenceValidatorConfig = DefaultConfig
+  private lazy val validator = new CachedTopicsExistenceValidator(kafkaConfig, validatorConfig)
```

Comment:
does this has to be private? It makes quite difficult to provide own validator?

Reply:
good point, changed to protected